src/test/kotlin/no/liflig/documentstore/ExampleSearchRepository.kt

```diff
@@ -4,36 +4,76 @@ package no.liflig.documentstore
 
 import kotlinx.serialization.UseSerializers
 import no.liflig.documentstore.dao.AbstractSearchRepository
+import no.liflig.documentstore.dao.AbstractSearchRepositoryWithCount
 import no.liflig.documentstore.dao.SerializationAdapter
 import no.liflig.documentstore.entity.VersionedEntity
 import org.jdbi.v3.core.Jdbi
 
-class ExampleQueryObject
+data class ExampleQueryObject(
+  val limit: Int? = null,
+  val offset: Int? = null,
+  val orderBy: OrderBy? = null,
+  val orderDesc: Boolean = false,
+)
+
+enum class OrderBy {
+  TEXT,
+  CREATED_AT,
+}
 
 class ExampleSearchRepository(
   jdbi: Jdbi,
   sqlTableName: String,
   serializationAdapter: SerializationAdapter<ExampleEntity>
-) : AbstractSearchRepository<ExampleId, ExampleEntity, ExampleQueryObject>(jdbi, sqlTableName, serializationAdapter) {
-  enum class OrderBy {
-    TEXT,
-    CREATED_AT,
-  }
-
+) :
+  AbstractSearchRepository<ExampleId, ExampleEntity, ExampleQueryObject>(
+    jdbi, sqlTableName, serializationAdapter
+  ) {
   override fun listByIds(ids: List<ExampleId>): List<VersionedEntity<ExampleEntity>> {
     TODO("Not yet implemented")
   }
 
   override fun search(query: ExampleQueryObject): List<VersionedEntity<ExampleEntity>> =
+    getByPredicate(
+      limit = query.limit,
+      offset = query.offset,
+      orderDesc = query.orderDesc,
+      orderBy =
+      when (query.orderBy) {
+        OrderBy.TEXT -> "data->>'text'"
+        OrderBy.CREATED_AT -> "createdAt"
+        null -> null
+      }
+    )
+}
+
+class ExampleSearchRepositoryWithCount(
+  jdbi: Jdbi,
+  sqlTableName: String,
+  serializationAdapter: SerializationAdapter<ExampleEntity>
+) :
+  AbstractSearchRepositoryWithCount<ExampleId, ExampleEntity, ExampleQueryObject>(
+    jdbi,
+    sqlTableName,
+    serializationAdapter,
+  ) {
+  override fun search(query: ExampleQueryObject): List<VersionedEntity<ExampleEntity>> {
     TODO("Not yet implemented")
+  }
 
-  fun search(limit: Int? = null, offset: Int? = null, orderBy: OrderBy? = null, orderDesc: Boolean = false) =
-    getByPredicate(
-      limit = limit, offset = offset, orderDesc = orderDesc,
-      orderBy = when (orderBy) {
+  override fun searchWithCount(
+    query: ExampleQueryObject
+  ): Pair<List<VersionedEntity<ExampleEntity>>, Long> {
```

Comment:
Anbefaler å lage en `data class` med feltene `entities` og `count` i stedet for å bruke `Pair`. Genriske klasser som `pair` er kjipe fordi du mister navn på feltene. Du vet ikke automatisk hva `second` er, men du skjønner fort `count`.

Reply:
Endra i fd30dce!